### PR TITLE
PullReq

### PR DESCRIPTION
**Pull request by Gaborori**

### CODE REVIEW

**Comment by Gaborori** [comment]
New description

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -13,6 +13,8 @@
   private double FAILURE_RATE = 0.0; //NOSONAR
 
   private int torpedoCount = 0;
+  
+  private int rand = new Random();
```
Comment (adamtorok96) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:17]: You must replace 'int' to Random

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+     throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (adamtorok96) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: Looks good :)

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
-    double r = generator.nextDouble();
+    double r = rand.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (adamtorok96) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:45]: Nice fix

**Review by adamtorok96 — CHANGES_REQUESTED** [review]

**Review by adamtorok96 — APPROVED** [review]
Approved

**Pull request merged**